src/main/scala/firrtl/transforms/ReplaceTruncatingArithmetic.scala

```diff
@@ -24,10 +26,22 @@ object ReplaceTruncatingArithmetic {
     */
   def onExpr(netlist: Netlist)(expr: Expression): Expression =
     expr.map(onExpr(netlist)) match {
+      // If an unsigned wrapping add/sub
       case orig @ DoPrim(Tail, Seq(e), SeqBIOne, tailtpe) =>
         netlist.getOrElse(we(e), e) match {
-          case DoPrim(Add, args, cs, _) => DoPrim(Addw, args, cs, tailtpe)
-          case DoPrim(Sub, args, cs, _) => DoPrim(Subw, args, cs, tailtpe)
+          case DoPrim(Add, args, cs, u: UIntType) => DoPrim(Addw, args, cs, tailtpe)
+          case DoPrim(Sub, args, cs, u: UIntType) => DoPrim(Subw, args, cs, tailtpe)
```

Comment:
I realize now that the spec is out of date about `sub` result types -- see #1378